src/Altinn.App.Api/Controllers/InstancesController.cs

```diff
@@ -1313,4 +1333,23 @@ await _instanceClient.UpdateDataValues(
             );
         }
     }
+
+    private async Task<InstantiationValidationResult> ValidationResultWithMessage(
+        InstantiationValidationResult validationResult,
+        string? language
+    )
+    {
+        if (String.IsNullOrEmpty(validationResult.CustomTextKey) || !String.IsNullOrEmpty(validationResult.Message))
+        {
+            return validationResult;
+        }
+
+        return new InstantiationValidationResult()
+        {
+            Valid = validationResult.Valid,
+            CustomTextKey = validationResult.CustomTextKey,
+            Message = await _translationService.TranslateTextKey(validationResult.CustomTextKey, language),
+            ValidParties = validationResult.ValidParties,
+        };
```

Comment:
I don't think it makes much sense to create a new object here. Immutable updates isn't worth a whole lot (when we don't use the object id for fast compare), and we will get funny behaviour if someone adds a new property without adding it to the mapping.

We could change ``InstantiationValidationResult`` to a record so we can use the ``with`` keyword if we need the immutable updates.

```c#
return validationResult with
{
    Message = await _translationService.TranslateTextKey(validationResult.CustomTextKey, language),
}
```